src/nvim/option_defs.h

```diff
@@ -151,26 +151,39 @@
 
 #define COCU_ALL        "nvic"          /* flags for 'concealcursor' */
 
-/* characters for p_shm option: */
-#define SHM_RO          'r'             /* readonly */
-#define SHM_MOD         'm'             /* modified */
-#define SHM_FILE        'f'             /* (file 1 of 2) */
-#define SHM_LAST        'i'             /* last line incomplete */
-#define SHM_TEXT        'x'             /* tx instead of textmode */
-#define SHM_LINES       'l'             /* "L" instead of "lines" */
-#define SHM_NEW         'n'             /* "[New]" instead of "[New file]" */
-#define SHM_WRI         'w'             /* "[w]" instead of "written" */
-#define SHM_A           "rmfixlnw"      /* represented by 'a' flag */
-#define SHM_WRITE       'W'             /* don't use "written" at all */
-#define SHM_TRUNC       't'             /* trunctate file messages */
-#define SHM_TRUNCALL    'T'             /* trunctate all messages */
-#define SHM_OVER        'o'             /* overwrite file messages */
-#define SHM_OVERALL     'O'             /* overwrite more messages */
-#define SHM_SEARCH      's'             /* no search hit bottom messages */
-#define SHM_ATTENTION   'A'             /* no ATTENTION messages */
-#define SHM_INTRO       'I'             /* intro messages */
-#define SHM_COMPLETIONMENU 'c'          // completion menu messages
-#define SHM_ALL         "rmfixlnwaWtToOsAIc" /* all possible flags for 'shm' */
```

Comment:
`option.c`, function `shortmess` contains `vim_strchr(p_shm, 'a')`, it is also the only other place where SHM_A is used. I would suggest:

1. Rename SHM_A to SHM_ALL_ABBREVIATIONS (SHM_ALL + _ABBREVIATIONS).
2. Add `SHM_ABBREVIATIONS = 'a',  ///< Use abbreviations from #SHM_ALL_ABBREVIATIONS` to the enum.
3. Use this constant in place of `'a'` in `shortmess` function.